File: packages/typescript/tests/system/navigation.test.ts
import { NavigateBackTool } from "alumnium";
import { describe } from "vitest";
import { baseIt } from "./helpers.ts";

describe("Navigation", () => {
  const it = baseIt.override("setup", async ({ setup, skip }) => {
    return async (options) => {
      const result = await setup(options);
      const { model } = result;

      if (model.provider === "mistralai")
        skip("Mistral needs more work on navigation");

      return result;
    };
  });

  it("navigate back uses history", async ({ expect, setup }) => {
    const { al, $ } = await setup({
      extraTools: [NavigateBackTool],
    });

    await $.navigate("https://the-internet.herokuapp.com");
    expect(await al.driver.url()).toBe("https://the-internet.herokuapp.com/");

    await al.do("open typos");
    expect(await al.driver.url()).toBe(
      "https://the-internet.herokuapp.com/typos",
    );

    await al.do("navigate back to the previous page");
    expect(await al.driver.url()).toBe("https://the-internet.herokuapp.com/");
  });
});

File: packages/python/examples/pytest/navigation_test.py
import pytest

from alumnium import Provider
from alumnium.tools import NavigateBackTool


def test_navigate_back_uses_history(al_factory, navigate):
    al = al_factory(extra_tools=[NavigateBackTool])
    if al.model.provider == Provider.MISTRALAI:
        pytest.xfail("Needs more work")

    navigate("https://the-internet.herokuapp.com")
    assert al.driver.url == "https://the-internet.herokuapp.com/"

    al.do("open typos")
    assert al.driver.url == "https://the-internet.herokuapp.com/typos"

    al.do("navigate back to the previous page")
    assert al.driver.url == "https://the-internet.herokuapp.com/"
